### PR TITLE
Train and validation splits of datasets.

### PR DESCRIPTION
**Pull request by AtanasGruev**
In Supplementary Materials 4.4 of the European Space Agency Anomaly Detection Benchmark (ESA-ADB) paper, training and validations splits of different mission phases are discussed, with the test set being fixed per mission dataset. On the other hand, when evaluating deep learning algorithms on different mission phases, validation splits are passed as arguments and used for hyperparameter tuning. This PR makes the point that a clear separation of training and validation sets can be conducted on the mission level, expanding beyond algorithms.

Concretely, the `train_val_dataset_split.py` can be run after the main preprocessing pipeline to obtain metadata for train / val splits of the default training sets, as well as to persist train / val splits to disk in accordance with TimeEval functionality. In turn, the user can prevent overwriting of the default `datasets.csv` used by TimeEval, and specify `--index_filename` to configure a target filepath for the (train, val)-Datasets. In that case, manual adjustment of `datasets.csv` would be due. 

Ultimately, the script is intended to complement the main preprocessing pipeline by extracting statistics on train / val splits and modifying `datasets.csv` -- that would enable easy adaptation of `mission1_experiments.py` and `mission2_experiments.py` to run on (train, val)-Datasets so as to tune hyperparameters for algorithms in general, to explore parameters based on dataset statistics, etc.

### CODE REVIEW

**Comment by AtanasGruev** [comment]
It might be a good idea to differentiate between `--src_index_filename` and `--dst_index_filename`. Currently, `timeeval.DatasetManager` loads and selects the datasets from the default `datasets.csv` which is inconvenient if it has thereafter been altered to include the (train, val)-Datasets for experiments. In the original commit, only the destination can be controlled via `--index_filename`.

**Review by Kotrix — APPROVED** [review]
